File: week02/11047.cpp
#include <iostream>
int main() {
    int n, k, i, coin[10], ans = 0;
    scanf("%d %d", &n, &k);
    for (i = 0; i < n; ++i) scanf("%d", &coin[i]);
    for (i = n - 1; i >= 0 && k != 0; --i) {
        ans += k / coin[i];
        k %= coin[i];
    }
    printf("%d", ans);
}

File: week02/11047.py
n, k = map(int, input().split())
coin = []; ans = 0
for i in range(n): coin.append(int(input()))
for i in range(n - 1, -1, -1):
    ans += k // coin[i]
    k %= coin[i]
    if (k == 0): break
print(ans)